app-header-layout/app-header-layout.html

```diff
@@ -63,44 +70,36 @@
         height: 100%;
       }
 
-      :host > ::content > app-header,
-      :host > ::content > [underlay-content] {
-        position: fixed;
-        top: 0;
-        right: 0;
-        left: 0;
+      :host > ::content > app-header {
+        @apply(--layout-fixed-top);
+
+        z-index: 1;
       }
 
-      :host([has-scrolling-region]) > ::content > app-header,
-      :host([has-scrolling-region]) > ::content > [underlay-content] {
+      :host([has-scrolling-region]) > ::content > app-header {
         position: absolute;
       }
 
+      #contentContainer {
+        /* Create a stacking context here so that all children appear below the header. */
+        position: relative;
+        z-index: 0;
+      }
+
       :host([has-scrolling-region]) > #contentContainer {
         @apply(--layout-fit);
 
         overflow-y: auto;
         -webkit-overflow-scrolling: touch;
       }
-
-      #contentContainer {
-        position: relative;
-
-        overflow-y: hidden;
-
-        @apply(--app-header-layout-content);
```

Comment:
I am guessing removing this is intentional since none of the demos are using it.  But just want to make sure.